api/src/main/java/jakarta/data/repository/CrudRepository.java

```diff
@@ -95,8 +113,11 @@ public interface CrudRepository<T, K> extends BasicRepository<T, K> {
      * <p>Non-matching entities are ignored and do not cause an error to be raised.</p>
      *
      * @param entities entities to update.
-     * @return the number of matching entities that were found in the database to update.
+     * @param <S> Type of the entities to update.
+     * @return an iterable containing the updated entities. In the case of Jakarta Data providers
+     *         that support Java records, this may return a different instance from the input,
+     *         preserving immutability.
```

Comment:
It is not acceptable to force retrieval from the database on update.  Please restore this to how it was, as follows:
```suggestion
     * @return the number of matching entities that were found in the database to update.
```